--- origsrc/e16-0.16.8.13/scripts/e_gen_menu	2008-03-22 11:20:18.000000000 -0500
+++ src/e16-0.16.8.13/scripts/e_gen_menu	2008-05-23 09:30:22.687500000 -0500
@@ -378,9 +378,9 @@
 		next if ($done{$d});
 		$done{$d} = 1;
 		print "Looking for epplets in $d\n" if $dbg ge 1;
-		@el = grep /\.epplet$/, ReadDir($d);
+		@el = grep /\.epplet\.exe$/, ReadDir($d);
 		foreach $e (@el) {
-			$e =~ s/\.epplet$//;
+			$e =~ s/\.epplet\.exe$//;
 			$i = "$EdirRoot/epplet_icons/$e.icon";
 			print FD "\"$e\" \"$i\" exec \"$d/$e.epplet\"\n";
 		}
--- origsrc/e16-0.16.8.13/src/misc.c	2008-05-01 06:33:22.000000000 -0500
+++ src/e16-0.16.8.13/src/misc.c	2008-05-23 08:21:39.375000000 -0500
@@ -320,7 +320,14 @@
    char                buf[1024];
    void               *h;
 
+#ifdef __CYGWIN__
+#ifndef RTLD_LOCAL
+#define RTLD_LOCAL 0
+#endif
+   Esnprintf(buf, sizeof(buf), "%s/cyg%s_%s.dll", ENLIGHTENMENT_LIB, lib, name);
+#else
    Esnprintf(buf, sizeof(buf), "%s/lib%s_%s.so", ENLIGHTENMENT_LIB, lib, name);
+#endif
    if (EDebug(1))
       Eprintf("ModLoad %s\n", buf);
    h = dlopen(buf, RTLD_NOW | RTLD_LOCAL);
