--- origsrc/e16-1.0.10/scripts/e_gen_menu	2011-08-24 15:37:33.000000000 -0500
+++ src/e16-1.0.10/scripts/e_gen_menu	2011-11-03 18:40:32.676348400 -0500
@@ -379,9 +379,9 @@ sub MakeEppsMenu {
 		next if ($done{$d});
 		$done{$d} = 1;
 		print "Looking for epplets in $d\n" if $dbg ge 1;
-		@el = grep /\.epplet$/, ReadDir($d);
+		@el = grep /\.epplet\.exe$/, ReadDir($d);
 		foreach $e (@el) {
-			$e =~ s/\.epplet$//;
+			$e =~ s/\.epplet\.exe$//;
 			$i = "$EdirRoot/epplet_icons/$e.icon";
 			print FD "\"$e\" \"$i\" exec \"$d/$e.epplet\"\n";
 		}
--- origsrc/e16-1.0.10/src/misc.c	2011-08-28 14:34:48.000000000 -0500
+++ src/e16-1.0.10/src/misc.c	2011-11-03 18:41:07.339608800 -0500
@@ -308,7 +308,11 @@ ModLoadSym(const char *lib, const char *
    char                buf[1024];
    void               *h;
 
+#ifdef __CYGWIN__
+   Esnprintf(buf, sizeof(buf), "%s/cyg%s_%s.dll", EDirLib(), lib, name);
+#else
    Esnprintf(buf, sizeof(buf), "%s/lib%s_%s.so", EDirLib(), lib, name);
+#endif
    if (EDebug(1))
       Eprintf("ModLoad %s\n", buf);
    h = dlopen(buf, RTLD_NOW | RTLD_LOCAL);
